core/src/main/java/bisq/core/btc/wallet/XmrWalletService.java

```diff
@@ -485,9 +486,21 @@ private MoneroWalletRpc startWalletRpcInstance(Integer port) {
     }
 
     private void backupWallets() {
+
+        // backup main wallet
         FileUtil.rollingBackup(walletDir, xmrWalletFile.getName(), 20);
         FileUtil.rollingBackup(walletDir, xmrWalletFile.getName() + ".keys", 20);
         FileUtil.rollingBackup(walletDir, xmrWalletFile.getName() + ".address.txt", 20);
+
+        // backup multisig wallets
+        File[] multisigWalletFiles = walletDir.listFiles();
+        for (int i = 0; i < multisigWalletFiles.length; i++) {
+            if (multisigWalletFiles[i].isFile() && multisigWalletFiles[i].getName().startsWith(MONERO_MULTISIG_WALLET_PREFIX)) {
```

Comment:
Actually, this will start a rolling backup of each "wallet file set" multiple times, once per each wallet file.

Since each wallet is comprised of 3 files, each starting with the same prefix, this code will start a rolling backup for all 3 files, once per each wallet file with that prefix.

This causes the behavior of wallet files being backed up within milliseconds of each other.

We need to start a rolling backup for each wallet set only once. One way to do it is by maintaining a set of prefix strings which have already been processed.

Reply:
I came up with the idea of getting the wallet files by a list of trade ids. It's also a bit more efficient than looping through all the wallet files.